src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -2832,6 +2844,202 @@ ImmutableArray<IArgumentOperation> makeArguments(Lazy<ImmutableArray<IArgumentOp
+
+                        bool leftIsNullable = ITypeSymbolHelpers.IsNullableType(operation.Value.Type);
+                        bool rightIsNullable = ITypeSymbolHelpers.IsNullableType(singleValueClause.Value.Type);
+                        bool isLifted = leftIsNullable || rightIsNullable;
+                        IOperation leftOperand = getSwitchValue();
+                        IOperation rightOperand = Visit(singleValueClause.Value);
+
+                        if (isLifted)
+                        {
+                            if (!leftIsNullable)
+                            {
+                                if (leftOperand.Type != null)
+                                {
+                                    // PROTOTYPE(dataflow): Consider using similar conversion for TryMakeNullable value and using the helper here
+                                    leftOperand = new ConversionOperation(leftOperand, ConvertibleConversion.Instance, isTryCast: false, isChecked: false,
+                                                                          semanticModel: null, leftOperand.Syntax, singleValueClause.Value.Type,
+                                                                          constantValue: default, isImplicit: true);
+                                }
+                            }
+                            else if (!rightIsNullable && rightOperand.Type != null)
+                            {
+                                // PROTOTYPE(dataflow): Consider using similar conversion for TryMakeNullable value and using the helper here
+                                rightOperand = new ConversionOperation(rightOperand, ConvertibleConversion.Instance, isTryCast: false, isChecked: false,
+                                                                       semanticModel: null, rightOperand.Syntax, operation.Value.Type,
+                                                                       constantValue: default, isImplicit: true);
+                            }
+                        }
+
+                        condition = new BinaryOperatorExpression(BinaryOperatorKind.Equals,
+                                                                 leftOperand,
+                                                                 rightOperand,
+                                                                 isLifted,
+                                                                 isChecked: false, 
+                                                                 isCompareText: false,
+                                                                 operatorMethod: null,
+                                                                 semanticModel: null,
+                                                                 singleValueClause.Value.Syntax,
+                                                                 ((Operation)operation).SemanticModel.Compilation.GetSpecialType(SpecialType.System_Boolean), 
```

Comment:
Consider getting the `SemanticModel` and `Compilation` once, in the .ctor, and storing in `readonly` fields.

Reply:
> Consider getting the SemanticModel and Compilation once, in the .ctor, and storing in readonly fields.

Will add a prototype comment for that 

---
In reply to: [184554290](https://github.com/dotnet/roslyn/pull/26395#discussion_r184554290) [](ancestors = 184554290)